sql/hive/src/main/scala/org/apache/spark/sql/hive/TableReader.scala

```diff
@@ -248,11 +249,16 @@ class HadoopTableReader(
         // SPARK-13709: For SerDes like AvroSerDe, some essential information (e.g. Avro schema
         // information) may be defined in table properties. Here we should merge table properties
         // and partition properties before initializing the deserializer. Note that partition
-        // properties take a higher priority here. For example, a partition may have a different
-        // SerDe as the one defined in table properties.
+        // properties take a higher priority here except for the Avro table properties
+        // to support schema evolution: in that case the properties given at table level will
+        // be used (for details please check SPARK-26836).
+        // For example, a partition may have a different SerDe as the one defined in table
+        // properties.
         val props = new Properties(tableProperties)
-        partProps.asScala.foreach {
-          case (key, value) => props.setProperty(key, value)
+        partProps.asScala.filterNot { case (k, _) =>
+          k == AvroTableProperties.SCHEMA_LITERAL.getPropName() && tableProperties.containsKey(k)
+        }.foreach { case (key, value) =>
+          props.setProperty(key, value)
```

Comment:
Let's revert to the original form like the following. It will make your PR content more clear.
```scala
-        }.foreach { case (key, value) =>
-          props.setProperty(key, value)
+        }.foreach {
+          case (key, value) => props.setProperty(key, value)
```

Reply:
Even when 2 lines above the other style looks good to avoid the 100 character limit for lines?

So do you prefer this?

```
          partProps.asScala.filterNot { case (k, _) =>
            k == AvroTableProperties.SCHEMA_LITERAL.getPropName() && tableProperties.containsKey(k)
          }.foreach {
            case (key, value) => props.setProperty(key, value)
          }
```

Reply:
I have chosen the consistent over keeping the old style. Or should I modify the `filterNot` body too?

Reply:
Yes, that one.